atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueService.java

```diff
@@ -761,18 +777,13 @@ public void truncateTables(final Set<TableReference> tablesToTruncate) {
                     @Override
                     public Void apply(Client client) throws Exception {
                         for (TableReference tableRef : tablesToTruncate) {
-                            if (shouldTraceQuery(tableRef)) {
-                                ByteBuffer recv_trace = client.trace_next_query();
-                                Stopwatch stopwatch = Stopwatch.createStarted();
-                                truncateInternal(client, tableRef);
-                                long duration = stopwatch.elapsed(TimeUnit.MILLISECONDS);
-                                if (duration > getMinimumDurationToTraceMillis()) {
-                                    log.error("Traced a call to " + tableRef.getQualifiedName() + " that took " + duration + " ms."
-                                            + " It will appear in system_traces with UUID=" + CassandraKeyValueServices.convertCassandraByteBufferUUIDtoString(recv_trace));
+                            run(client, tableRef, new Action<Void>() {
+                                @Override
+                                public Void run() throws TException {
+                                    truncateInternal(client, tableRef);
```

Comment:
I think because of how truncateInternal works the tracing will be off here.  @clockfort do we ever want to trace a truncate table call?